backend/src/socket/trace/base.trace.ts

```diff
@@ -17,6 +22,10 @@ export const createTrace = async (
   ) {
     throw new Error("Invalid board project or user in Trace Context");
   } else {
+    // check tracing enabled
+    if (!board.permissions.allowTracing) {
```

Comment:
Instead of querying the board each time, can we update `TraceContext` to have a boolean field called `enabled`? 

This way we can just do `if (input.trace.enabled) await createTrace(...)` inside each of the tracing events.